common/task_executor/src/lib.rs

```diff
@@ -179,6 +179,46 @@ impl TaskExecutor {
         }
     }
 
+    /// Run a task to completion on the runtime and return the result.
+    /// The task is wrapped in an exit future and is shutdown on receiving the exit signal.
+    ///
+    /// Returns `None` if the task is shutdown before completion or if the runtime has been dropped.
+    ///
+    /// Note: This method **will panic** if called from an async context.
+    pub fn block_on<V>(&self, task: impl Future<Output = V>, name: &'static str) -> Option<V> {
+        let exit = self.exit.clone();
+        let log = self.log.clone();
+
+        if let Some(int_gauge) = metrics::get_int_gauge(&metrics::ASYNC_TASKS_COUNT, &[name]) {
```

Comment:
Over in https://github.com/sigp/lighthouse/pull/2522/commits/4dfd853ac7b45df954b75094217bbc718c2d29e6, I did a refactor of `spawn_blocking_handle` for this reason:

> I noticed that the `TaskExecutor::spawn_blocking_handle` function would fail to spawn tasks if it were unable to obtain handles to some metrics (this can happen if the same metric is defined twice). It seemed that a more sensible approach would be to keep spawning tasks, but without metrics. To that end, I refactored the function so that it would still function without metrics. There are no other changes made.

It seems I missed `spawn_handle` though :( Do you mind shifting this implementation (`block_on`) around so it's like `spawn_blocking_handle` and doesn't fail to spawn tasks if there is a duplicate in metrics definitions, please? :pray:

Reply:
Fixed. I shifted the other spawns also to work even with duplicates.